### PR TITLE
Add documentation for Prometheus disk I/O metrics

### PR DESCRIPTION
**Pull request by ggilmore**
This post provides an overview of the newly added Prometheus disk I/O metrics for the Zoekt, Git Server, and Searcher services. 

It explains how these metrics can help identify potential performance issues and guide scaling decisions. It also includes links to relevant pull requests and resources for interpreting these metrics.

### CODE REVIEW

**Comment by ggilmore** [comment]
This is an  adaptation of the scale testing post https://sourcegraph.slack.com/archives/C040LV3PS4C/p1667946291146679

File: content/departments/engineering/dev/tools/observability/disk-metrics.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+- Determine how to scale a service (horizontally or vertically). For example, if a service's read / write request load is approaching the maximum IOPS capacity of the disk it is using, it may be necessary to add more replicas to distribute the load across multiple disks.
+
+For more information about the implementation of these metrics, see the following pull requests on GitHub:
+
+- [sourcegraph/pull/43070](https://github.com/sourcegraph/sourcegraph/pull/43070)
+- [sourcegraph/pull/43768](https://github.com/sourcegraph/sourcegraph/pull/43768)
+- [sourcegraph/pull/43877](https://github.com/sourcegraph/sourcegraph/pull/43877)
+- [sourcegraph/mountinfo](https://github.com/sourcegraph/mountinfo)
+
+For more information on how to interpret these metrics, see our [monitoring reference](https://docs.sourcegraph.com/admin/observability/dashboards#zoekt-data-disk-i-o-metrics-only-available-on-sourcegraph-internal-instances), as well as the following resources:
```
Comment (bobheadxi) [inline content/departments/engineering/dev/tools/observability/disk-metrics.md]: Should this link be https://docs.sourcegraph.com/admin/observability/dashboards#zoekt-data-disk-i-o-metrics ?

```suggestion
For more information on how to interpret these metrics, see our [monitoring reference](https://docs.sourcegraph.com/admin/observability/dashboards#zoekt-data-disk-i-o-metrics), as well as the following resources:
```

Comment (ggilmore) [reply content/departments/engineering/dev/tools/observability/disk-metrics.md]: You're correct!

**Review by daxmc99 — APPROVED** [review]
Wonderful! 
Looks like a good reference for disk metrics & sourcegraph performance

**Pull request merged**